### PR TITLE
feat(QUI-86): setup simple indexer RPC endpoint

### PR DESCRIPTION
**Pull request by L8D**

### CODE REVIEW

**Comment by L8D** [comment]
* **#56** <a href="https://app.graphite.dev/github/pr/Quible-Network/quible-node/56?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Quible-Network/quible-node/56?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/main.rs [hunk 1]
```diff
@@ -311,6 +311,20 @@ async fn propose_block(
...
+            };
+
+            let owner = match &pubkey_script[..] {
+                [TransactionOpCode::Dup, TransactionOpCode::Push { data: address_vec }, TransactionOpCode::EqualVerify, TransactionOpCode::CheckSigVerify] => {
+                    hex::encode(address_vec.as_slice())
+                }
+                _ => "".to_string(),
+            };
+
+            dbg!("WHATWHATWHAT");
```
Comment (c-gamble) [inline src/main.rs:326]: do we want this in production

File: src/main.rs [hunk 2]
```diff
@@ -1141,4 +1223,84 @@ mod tests {
...
+    async fn fetches_unspent_value_outputs() -> anyhow::Result<()> {
+        // Initialize SurrealDB
+        let db = any::connect("memory").await?;
+        db.use_ns("quible").use_db("quible_node").await?;
+        db::schema::initialize_db(&db).await?;
+
+        let db_arc = Arc::new(db);
+
+        let server_signer_key = k256::ecdsa::SigningKey::from_slice(&hex_literal::hex!(
+            "ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
```
Comment (c-gamble) [inline src/main.rs:1237]: should this be env variable for security

File: src/main.rs (hunk 1 above)
Comment (L8D) [reply src/main.rs:326]: Good catch. This was from some debugging but it is no longer needed.

File: src/main.rs (hunk 2 above)
Comment (L8D) [reply src/main.rs:1237]: It is an env variable. This is just a unit test. This is the default private key used by the `hardhat node` command

File: src/main.rs [hunk 3]
```diff
@@ -1141,4 +1221,84 @@ mod tests {
...
+        ))?;
+
+        let server_addr = run_derive_server(
+            server_signer_key.to_bytes().as_slice().try_into()?,
+            &db_arc,
+            0,
+        )
+        .await?;
+        let url = format!("http://{}", server_addr);
```
Comment (c-gamble) [inline src/main.rs:1244]: This will be error prone once we have an SSL certificate on our domain, but I think it's fine for now

**Review by c-gamble — APPROVED** [review]
Small change for the future when we point the node to an https: domain but for now it's fine

**Pull request merged**